### ISSUE TITLE
Make some simple modifications that will support use of full domains and ports in URLs

### ISSUE THREAD

**Issue by rberlind** [microservices-demo/microservices-demo#686]
I'm running the Socks Shop apps on the Apcera platform in which I currently need to use fully qualified DNS names having the form \<service\>.apcera.local rather than simple host names and ports different from 80 for some technical reasons which I would rather not go into here.  This has caused me some problems.

I was able to overcome most of the problems by either setting certain environment variables or adding extra command line arguments to the start commands Apcera uses when launching the apps.  That was mainly possible because most of the apps had actually provided mechanisms to set domain variables or override host variables. But there are still 2 issues that require code changes.

1) queue-master has the rabbitmq host hard-coded to "rabbitmq" in RabbitMqConfiguration.java and does not include the spring.rabbitmq.host setting in application.properties.  Specifically, queue-master uses "CachingConnectionFactory connectionFactory = new CachingConnectionFactory("rabbitmq");".  I would like to see queue-master changed to do what shipping does, namely add:
@Value("${spring.rabbitmq.host}")
private String host;
to RabbitMqConfiguration.java and add spring.rabbitmq.host=rabbitmq to application.properties.  This would then allow me to add  "--spring.rabbitmq.host=rabbitmq.apcera.local" to the queue-master start command as I did for shipping.  (See below.)

2) There is also an issue with some URL manipulation done in the customer-orders.html page in the front-end app.  Specifically, line 180 strips off the beginning of the full URL with selfRef.replace("http://orders", "").  But since I am using "orders.apcera.local:8080", this leaves me with order=.apcera.local:8080/orders/<order_id> and no details appear on the customer-order.html page after clicking the View button for an order.  I propose that that code be modified to `selfRef.replace(/http:\/\/(.*)\/orders/, "/orders")`.  That would then better tolerate situations like mine where a full domain and port are used.  Note that this modified code uses regex and escapes "/" where needed.

For reference, here are the other changes I made to support a full domain and port in URLs without code changes:

1) Set PORT environment variables for catalogue, cart, orders, user, payment, shipping, and queue-master to 8080.
2) Set ZIPKIN environment variables to include "zipkin.apcera.local" instead of "zipkin".
3) front-end: added "-- --domain=apcera.local:8080" to start command.
4) catalogue: added "-DSN=catalogue_user:default_password@tcp(catalogue-db.apcera.local:3306)/socksdb" to start command.
5) carts: set DB environment variable to "carts-db.apcera.local".
6) orders: set DB enviroment variable to "orders-db.apcera.local" and DOMAIN environment variable to "apcera.local:8080" (for use in OrdersConfigurationProperties.java).
7) user: set environment variables HATEAOS to "user.apcera.local:8080" and MONGO_HOST to "user-db.apcera.local:27017".
8) shipping: added "--spring.rabbitmq.host=rabbitmq.apcera.local" to start command

Thanks so much for your help with this.  I will try and submit pull requests with actual code this weekend or early next week.

Roger Berlind
Apcera

**Comment by rberlind** [comment]
The change I asked for in front-end was made via [front-end 87](https://github.com/microservices-demo/front-end/pull/87), but I realized later that I had left out a "}" and submitted [front-end 89](https://github.com/microservices-demo/front-end/pull/89) which is still pending.

I also submitted [queue-master 17](https://github.com/microservices-demo/queue-master/pull/17) but have not seen any action on that one.  The Travis coveralls test fails, but I think that is because my fork does not have your coveralls token.

Finally, I reported [issue 46](https://github.com/microservices-demo/user/issues/46) asking if the user app could be recompiled with Go 1.7.5 instead of Go 1.7.0 which I suspect was used.

I would love to see the 2 open pull requests merged and the user issue addressed.  At that point, I plan on submitting a pull request to the documentation and deployment that would add Apcera as an additional deployment platform for Socks Shop.  I would like to do this before DockerCon which starts April 17 since Apcera will be showing our version of the Socks Shop demo there.

Thanks
Roger
